File: Valid_Sudoku/valid_sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        seen_rows = [set() for _ in range(9)]
        seen_cols = [set() for _ in range(9)]
        seen_box = [set() for _ in range(9)]

        for i in range(9):
            for j in range(9):
                curr = int(board[i][j]) if board[i][j] != "." else -1
                if curr != -1:
                    if curr in seen_rows[i] or curr in seen_cols[j] or curr in seen_box[(i // 3) * 3 + j // 3]:
                        return False
                seen_rows[i].add(curr)
                seen_cols[j].add(curr)
                seen_box[(i // 3) * 3 + j // 3].add(curr)


        return True                

File: Valid_Sudoku/valid_sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        std::vector<std::unordered_set<int>> seen_rows(9);        
        std::vector<std::unordered_set<int>> seen_cols(9);
        std::vector<std::unordered_set<int>> seen_box(9);

        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                int curr;
                if (board[i][j] != '.') {
                    curr = board[i][j];
                } else {
                    continue;
                }
                if (seen_rows[i].count(curr) || seen_cols[j].count(curr) || seen_box[(i / 3) * 3 + (j / 3)].count(curr)) {
                    return false;
                }
                seen_rows[i].insert(curr);
                seen_cols[j].insert(curr);
                seen_box[(i / 3) * 3 + (j / 3)].insert(curr);
            }
        }

        return true;
    }
};
